apm-application-toolkit/apm-toolkit-webflux/src/main/java/org/apache/skywalking/apm/toolkit/webflux/WebFluxSkyWalkingOperators.java

```diff
@@ -0,0 +1,120 @@
+import reactor.util.context.Context;
+
+import java.util.concurrent.Callable;
+import java.util.function.Consumer;
+
+/**
+ * WebFlux operators that are capable to reuse tracing context from Reactor's Context.
+ */
+public final class WebFluxSkyWalkingOperators {
+
+    private WebFluxSkyWalkingOperators() {
+        throw new IllegalStateException("You can't instantiate a utility class");
+    }
+
+    /**
+     * Wraps a runnable with a span.
+     *
+     * @param signalType - Reactor's signal type
+     * @param runnable   - lambda to execute within the tracing context
+     * @return consumer of a signal
+     */
+    public static Consumer<Signal<?>> withSpanInScope(SignalType signalType, Runnable runnable) {
+        return signal -> {
+            if (signalType != signal.getType()) {
+                return;
+            }
+            withSpanInScope(runnable).accept(signal);
+        };
+    }
+
+    /**
+     * Wraps a runnable with a span.
+     *
+     * @param signalType - Reactor's signal type
+     * @param consumer   - lambda to execute within the tracing context
+     * @return consumer of a signal
+     */
+    public static Consumer<Signal> withSpanInScope(SignalType signalType, Consumer<Signal> consumer) {
+        return signal -> {
+            if (signalType != signal.getType()) {
+                return;
+            }
+            withSpanInScope(signal.getContext(), () -> consumer.accept(signal));
+        };
+    }
+
+    /**
+     * Wraps a runnable with a span.
+     *
+     * @param runnable - lambda to execute within the tracing context
+     * @return consumer of a signal
+     */
+    public static Consumer<Signal> withSpanInScope(Runnable runnable) {
+        return signal -> {
+            Context context = signal.getContext();
+            withSpanInScope(context, runnable);
+        };
+    }
+
+    /**
+     * Wraps a runnable with a span.
+     *
+     * @param context  - Reactor context that contains the tracing context
+     * @param runnable - lambda to execute within the tracing context
+     */
+    public static void withSpanInScope(Context context, Runnable runnable) {
+        runnable.run();
+    }
+
+    /**
+     * Wraps a callable with a span.
+     *
+     * @param context  - Reactor context that contains the tracing context
+     * @param callable - lambda to execute within the tracing context
+     * @param <T>      callable's return type
+     * @return value from the callable
+     */
+    public static <T> T withSpanInScope(Context context, Callable<T> callable) {
+        try {
+            return callable.call();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        } 
+    }
+
+    public static <T> T withSpanInScope(ServerWebExchange serverWebExchange, Callable<T> callable) {
+        try {
+            return callable.call();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
```

Comment:
Why do we through an IllegalStateException from any caught exception? This seems wrong.

Reply:
If throwing directly will cause trouble for the use of lambda, I have changed to the `sneakyThrow `

Reply:
I mean, you don't need to catch? As a wrapper, it should declare the same `throw` as the original.